### PR TITLE
Feature/add fork pr

### PR DESCRIPTION
**Pull request by shaharblanksela**
Add new pipelines:

1. Create branch from forked repo - the pipeline will run manually.  
   Need to specify 2 parameters (contributor name & the desired branch comes from the PR) - tested on the 
   custom repo I’ve created in my account. I needed to use PAT for the pipeline to work, I saw you already have PAT in 
   amplication secrets so I took that as ref.

3. Notification from PR to Discord - (Alert when new PR is opened. reopened, closed) tested and can be seen on the GitHub notifications channel.

### CODE REVIEW

**Comment by souravjain540** [comment]
Hey @mshidlov 👋 

Can you please review this PR?

File: .github/workflows/fork-to-branch.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  workflow_dispatch:
+    inputs:
+      source-branch:
+        type: string
+        description: The name of the branch from forked-repo.
+        required: true
+      contributor-name:
+        type: string
+        description: write the contributor github username
+        required: true
```
Comment (mshidlov) [inline .github/workflows/fork-to-branch.yaml:13]: This suppose to run automatically when a fork PR is opened and update the branch when the PR is updated

File: .github/workflows/fork-to-branch.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+
+      - name: Check if branch exists
+        run: |
+          existed_in_repo=$(git ls-remote --heads origin feature/${{ github.event.inputs.source-branch }})
```
Comment (mshidlov) [inline .github/workflows/fork-to-branch.yaml:26]: the branch name can start with different prefixes beside `feature/`

File: .github/workflows/fork-to-branch.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            git push origin --delete feature/${{ github.event.inputs.source-branch }}
+            echo "branch feature/${{ github.event.inputs.source-branch }} deleted" 
+          fi
+  
+      - name: Git remote add forked repo
+        run: |
+          git remote add upstream https://github.com/${{ github.event.inputs.contributor-name }}/amplication.git
+          git fetch --all
+          git checkout upstream/${{ github.event.inputs.source-branch }}
+          git checkout -b feature/${{ github.event.inputs.source-branch }}
```
Comment (mshidlov) [inline .github/workflows/fork-to-branch.yaml:40]: branches not always start with `feature/`

File: .github/workflows/fork-to-branch.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          git push --set-upstream origin feature/${{ github.event.inputs.source-branch }}
```
Comment (mshidlov) [inline .github/workflows/fork-to-branch.yaml:41]: we can use PR # and create branch `pr/${{pull-request.id}}`

File: .github/workflows/new-pr.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    types:
+      - opened
+      - reopened
+      - closed
+
+env:
+  PR_PROJECT_ID: 2
+  ORG: amplication
+  OPENED_REOPEND_STATUS: New
+  CLOSED_STATUS: Closed
```
Comment (mshidlov) [inline .github/workflows/new-pr.yaml:13]: This is not needed and can be removed

File: .github/workflows/new-pr.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    if: github.event_name == 'pull_request' && github.event.action == 'closed'
+    steps:
+      - name: 'Moved PR to "Closed"'
+        uses: leonsteinhaeuser/project-beta-automations@v1.0.2
+        with:
+          gh_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          organization: ${{ env.ORG }}
+          project_id: ${{ env.PR_PROJECT_ID }}
+          resource_node_id: ${{ github.event.pull_request.node_id }}
+          status_value: ${{ env.CLOSED_STATUS }}
```
Comment (mshidlov) [inline .github/workflows/new-pr.yaml:47]: This is not needed and can be removed

File: .github/workflows/new-pr.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    if: github.event_name == 'pull_request' && github.event.action == 'opened' || github.event.action == 'reopened'
+    steps:
+      - name: 'Move pr to "New"'
+        uses: leonsteinhaeuser/project-beta-automations@v1.0.2
+        with:
+          gh_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          organization: ${{ env.ORG }}
+          project_id: ${{ env.PR_PROJECT_ID }}
+          resource_node_id: ${{ github.event.pull_request.node_id }}
+          status_value: ${{ env.OPENED_REOPEND_STATUS }}
```
Comment (mshidlov) [inline .github/workflows/new-pr.yaml:28]: This is not needed and can be removed

**Comment by mshidlov** [comment]
Thank you @shaharblanksela it looks good I'm merging it

**Review by mshidlov — APPROVED** [review]

**Pull request merged**